<asm/types.h> doesn't define __u64 in __STRICT_ANSI__ (-ansi) mode.  In
order to use <linux/cdrom.h> in this mode we need to fix this file not to
assume __u64.

Update 2005-05-04 - gotom.

--- include.orig/asm-i386/byteorder.h	2005-04-21 09:03:16.000000000 +0900
+++ include/asm-i386/byteorder.h	2005-05-03 14:25:46.000000000 +0900
@@ -25,6 +25,10 @@
 	return x;
 }
 
+#define __arch__swab32(x) ___arch__swab32(x)
+
+#if !defined (__STRICT_ANSI__)
+
 static __inline__ __attribute_const__ __u64 ___arch__swab64(__u64 val)
 { 
 	union { 
@@ -48,10 +52,11 @@ static __inline__ __attribute_const__ __
    convert it into rotation or exhange.  */
 
 #define __arch__swab64(x) ___arch__swab64(x)
-#define __arch__swab32(x) ___arch__swab32(x)
 
 #define __BYTEORDER_HAS_U64__
 
+#endif /* !__STRICT_ANSI__ */
+
 #endif /* __GNUC__ */
 
 #include <linux/byteorder/little_endian.h>
